### ISSUE TITLE
offline-ssm-provider: please clarify intended behavior or update docs to state

### ISSUE THREAD

**Issue by bitsofinfo** [CoorpAcademy/serverless-plugins#152]
Unfortunately this plugin now has me confused and raises a larger issue. 

Maybe I'm mis-understanding what functionality this plugin is supposed to provide.

My impression was that this plugin provided the following:

**I thought that if you had a local .ssm file when running serverless-offline it would always use the value in .ssm... regardless if one existed and was accessible in the real AWS SSM store.**

However it appears that the .ssm file will ONLY be used when either you don't have access to the real value in AWS SSM, it can't be found in AWS SSM or you are truely "offline" don't have any network connectivity. If you do have access to the live AWS SSM value, that will be used and not the local `.ssm` one (when in serverless-offline) mode.

**Comment by silouone** [comment]
Great question @bitsofinfo — the behavior you observed (live AWS SSM taking precedence when reachable, falling back to the local `.ssm` file otherwise) is by design, as the provider acts as a fallback rather than an unconditional override; if you want the local values to always win for non-production secrets, the reliable approach is to keep those keys out of the real SSM store for your offline stage. We'll note this clarification in the docs 🙏

**Comment by bitsofinfo** [comment]
omg, 5 years later. hilarous.

**Comment by silouone** [comment]
@bitsofinfo yup sorry aha was left alone for too long, did a bit of a major updating/cleaning :)
